assets/src/blocks/CarouselHeader/CarouselHeaderEditor.js

```diff
@@ -50,30 +63,57 @@ export const CarouselHeaderEditor = ({ setAttributes, attributes }) => {
     goToSlide(currentSlide > lastSlide ? 0 : currentSlide, true);
   }
 
+  const needsMigration = slides.some(slide => !!slide.image && !slide.image_srcset)
+  const migratedSlides = useSelect(select=> slides && slides.map(slide => {
+    if (!needsMigration) {
```

Comment:
If it doesn't need migration, it will map to a bunch of nulls. That's not a problem because in that case the result is not used.